workspace-grid@mathematical.coffee.gmail.com/extension.js

```diff
@@ -976,23 +997,24 @@ function overrideWorkspaceDisplay() {
     wvStorage._init = WorkspacesView.WorkspacesView.prototype._init;
     WorkspacesView.WorkspacesView.prototype._init = function () {
         wvStorage._init.apply(this, arguments);
-        Main.overview.connect('scroll-event', Lang.bind(this, function _horizontalScroll(actor, event) {
-                // same as the original, but for LEFT/RIGHT
-                // if (!actor.mapped)
-                //     return false;
-                let wsIndex =  global.screen.get_active_workspace_index();
-
-                switch (event.get_scroll_direction()) {
-                    case Clutter.ScrollDirection.UP:
-                        global.screen.workspace_grid.actionMoveWorkspace(wsIndex-1);
-                        return Clutter.EVENT_STOP;
-                    case Clutter.ScrollDirection.DOWN:
-                        global.screen.workspace_grid.actionMoveWorkspace(wsIndex+1);
-                        return Clutter.EVENT_STOP;
-                }
+        // Main.overview.connect('scroll-event', Lang.bind(this, _horizontalScroll));
+        Main.overview.connect('scroll-event', Lang.bind(this, _scrollHandler));
```

Comment:
Why not define two different handlers and choose the right one according to the value of the "scroll-direction" option?

Reply:
I was working on an integration with the other extension, Dash to Dock, and I came across this problem. This decision eventually made the code needed for integration reduced from 15 lines (the handlers) to only 4 (just call actionMoveWorkspace).

Reply:
OK then, since I can't think of any better way to achieve this and you already tested it with another extension let's keep it this way, but please add a comment explaining this, so that nobody breaks it in the future.